mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -100,19 +100,26 @@ class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     val outputSchema = transformSchema(dataset.schema)
-    val hashUDF = udf { terms: Seq[_] =>
-      val numOfFeatures = $(numFeatures)
-      val isBinary = $(binary)
-      val termFrequencies = mutable.HashMap.empty[Int, Double].withDefaultValue(0.0)
-      terms.foreach { term =>
-        val i = indexOf(term)
-        if (isBinary) {
+    val localNumFeatures = $(numFeatures)
+
+    val hashUDF = if ($(binary)) {
```

Comment:
I guess I'm saying, is it worth duplicating the UDF definition to lift this check out? just make a local `val binary = $(binary)` if you want to avoid referencing the property every time.

Reply:
ok, I misunderstood the comments